### PR TITLE
Provide crypto.KeyManager through extension.Registry

### PR DESCRIPTION
**Pull request by RJPercival**
Allows crypto keys to be obtained from different sources, depending on the registry implementation.

### CODE REVIEW

File: server/sequencer_manager_test.go [hunk 1]
```diff
@@ -169,8 +170,19 @@ func mockStorageProviderForSequencer(mockStorage storage.LogStorage) testonly.Ge
 	}
 }
 
-func registryForSequencer(mockStorage storage.LogStorage) extension.Registry {
-	return testonly.NewRegistryWithLogProvider(mockStorageProviderForSequencer(mockStorage))
+func mockKeyManagerProviderForSequencer(mockKM crypto.KeyManager) testonly.GetKeyManagerFunc {
+	return func(id int64) (crypto.KeyManager, error) {
+		if mockKM == nil {
+			return nil, fmt.Errorf("mock KeyManager is nil")
```
Comment (AlCutter) [inline server/sequencer_manager_test.go]: can just use errors.New(...) here since there are no args.

**Comment by AlCutter** [comment]
Did you mean to move the `testonly/keys.go` under `crypto/`?
You'll probably need a `crypto/testonly` package if you want to do that (currently you have two packages in the `crypto/` dir)

File: server/sequencer_manager_test.go (hunk 1 above)
Comment (RJPercival) [inline server/sequencer_manager_test.go]: Done.

**Comment by RJPercival** [comment]
Travis is green.

**Comment by RJPercival** [comment]
Rebased following merge of PR #387, which addressed the import cycle. PTAL.

**Pull request merged**